central/sensor/service/connection/ratelimit_manager.go

```diff
@@ -0,0 +1,113 @@
+package connection
+
+import (
+	"math"
+	"time"
+
+	"github.com/stackrox/rox/central/sensor/service/connection/ratetracker"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/grpc/ratelimit"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const (
+	boostInitSyncRateLimit = 50
+
+	defaultTopCandidateClusters = 3
+	defaultRatePeriod           = 10 * time.Minute
+)
+
+type rateLimitManager struct {
+	mutex sync.Mutex
+
+	initSyncMaxSensors int
+	initSyncSensors    set.StringSet
+
+	msgRateLimiter ratelimit.RateLimiter
+	msgRateTracker ratetracker.ClusterRateTracker
+}
+
+// newRateLimitManager creates an rateLimitManager with max sensors
+// retrieved from env variable, ensuring it is non-negative.
+func newRateLimitManager() *rateLimitManager {
+	maxSensors := env.CentralMaxInitSyncSensors.IntegerSetting()
```

Comment:
I wonder if this should be here or on the caller

Reply:
if `newRateLimitManager` accept rate limiter then we can test it with mock implementation so no sleep in tests will be required